packages/data-context/src/sources/ProjectDataSource.ts

```diff
@@ -43,15 +49,13 @@ export class ProjectDataSource {
       return specs
     }
 
-    return specs.filter((spec) => spec.specType === specType)
-  }
+    project.specs = !specType ? specs : specs.filter((spec) => spec.specType === specType)
```

Comment:
Wait, why are we mutating `project.specs`?

Reply:
the `currentProject.specs` field was never being set, so they had to be set somewhere. I'll revert to just using `findSpecs` for the `getCurrentSpecByAbsolute` function. There will be some changes to how specs are managed here soon, we can tackle the inefficiencies soon